pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -371,7 +378,7 @@ func (hdl userDataHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if appConfig == nil {
 			log.Errorf("No AppNetworkConfig for %s",
 				anStatus.Key())
-			w.WriteHeader(http.StatusInternalServerError)
+			w.WriteHeader(http.StatusServiceUnavailable)
```

Comment:
That is clearly the wrong error.
503 Service Unavailable The server is currently unable to handle the request due to a temporary overload or scheduled maintenance, which will likely be alleviated after some delay. 

They are specified in an RFC, and we should follow that as closely as possible.